src/test/scala/org/broadinstitute/dsde/agora/server/webservice/PermissionIntegrationSpec.scala

```diff
@@ -494,6 +494,37 @@ class PermissionIntegrationSpec extends FlatSpec with RouteTest with ScalatestRo
       }
   }
 
+  "Agora" should "set the `public` field to true for a public entity" in {
+
+    patiently(permissionBusiness.insertEntityPermission(agoraEntity2, owner2.get,
+      AccessControl(AccessControl.publicUser, AgoraPermissions(AgoraPermissions.Read))))
+
+    Get(ApiUtil.Methods.withLeadingVersion + "/" + agoraEntity2.namespace.get + "/" +
+      agoraEntity2.name.get + "/" + agoraEntity2.snapshotId.get) ~>
+      methodsService.querySingleRoute ~> check {
+
+      val rawJs = responseAs[JsObject]
+      val entity = AgoraApiJsonSupportWithManagerAndPublicRead.AgoraEntityFormatWithManagerAndPublicRead.read(rawJs)
```

Comment:
I'm not too thrilled about the diverging JSON handling between test and production. I think it's possible for us to mess up `AgoraEntityFormat` and still have some of these tests pass.

I was thinking it might be nice to have `managers` and `public` set on _all_ Agora responses, and encapsulate the setting of those fields in `AgoraEntity` itself. I know, I know, OO is lame and all, but returning a different data structure between endpoints is becoming a bit difficult to deal with.

Wildly out of scope for these PRs, of course.

Reply:
I agree, this is messy - also agree it's out of scope for this PR.

We already set public on method lists (don't think we do it on config lists). Only concern about setting public and managers everywhere is the extra db queries ... it would probably be a perfectly acceptable performance hit, we'd just need to ensure it's acceptable.